rllib/agents/bandits/bandit_torch_policy.py

```diff
@@ -81,14 +78,14 @@ def make_model_and_action_dist(policy, obs_space, action_space, config):
 
     # TODO: Have a separate model catalogue for bandits
     if exploration_config:
-        if exploration_config["type"] == TS_PATH:
+        if exploration_config["type"] == "ThompsonSampling":
```

Comment:
maybe make this backwards compatible?

Reply:
Not sure this would work: The exploration type is now a single "ThompsonSampling" string as we moved the `ThompsonSampling` class into our built-in RLlib `utils/exploration` package.

The two supported exploration types are pretty much hard-coded into these two bandit Trainers (UCB and TS).
Also, the old `TS_PATH` ( -> ThompsonSampling path) does no longer exist as it was inside the contrib folder.

Reply:
OK, this may not be the right level to do backward compat messaging. Can you make sure that users code from previous ray versions will automatically raise an actionable error message to help them migrate to new changes?

Reply:
This is done through the added "location" check. The new `test_bandits` test case makes sure this works and users will get a deprecation message when they do stuff like:

```
tune.run("contrib/LinTS")

# OR

from ray.rllib.contrib.bandits.agents.lin_ts import LinTSTrainer
```